### ISSUE TITLE
Add a closure anomaly

### ISSUE THREAD

**Issue by simplesmiler** [ufocoder/javascript.anomaly#3]
``` js
for (var i = 0; i < 2; i++) {
  setTimeout(function() {
    console.log(i);
  }, 100);
}

// 2
// 2
```

**Comment by ufocoder** [comment]
Reference anomaly was added by b88fad17b791fa85dddb39ec94111a5768e6ef16 commit. Thanks for your enhancement

**Comment by simplesmiler** [comment]
Great, but it will output `3, 3, 3`, not `2, 2, 2`.
JavaScript is fun, I know :wink:

**Comment by ufocoder** [comment]
@simplesmiler Oh.. I was hurried and made a typo. Fixed by 153309b231d6ab7dbaefa288b42f1c62c8bda78f. Thanks for your feedback again
